Detectors/FIT/common/simulation/src/Digitizer.cxx

```diff
@@ -18,26 +18,83 @@
 #include <algorithm>
 #include <cassert>
 #include <iostream>
+#include <optional>
 
 using namespace o2::fit;
 //using o2::fit::Geometry;
 
 ClassImp(Digitizer);
 
+double sinc(const double x)
+{
+  if (x == 0)
+    return 1;
+  return sin(x) / x;
+}
+double signalForm_i(double x)
+{
+  return -(exp(-0.83344945 * x) - exp(-0.45458 * x)) * (x >= 0) / 7.8446501; // Maximum should be 7.0/250 mV
+};
+
 double get_time(const std::vector<double>& times, double signal_width)
 {
   TH1F hist("time_histogram", "", 1000, -0.5 * signal_width, 0.5 * signal_width);
-  auto signalForm = [](double x) {
-    return -(exp(-0.83344945 * x) - exp(-0.45458 * x));
-  };
-  for (auto time : times)
+  TH1F histsum("time_sum", "", 1000, -0.5 * signal_width, 0.5 * signal_width);
```

Comment:
The method ``get_time`` is called in a tight loop and for every such a call you will create/fill/delete 3 large histos. This is terribly inefficient. Since the ``signal_width`` parameter is always the same, you can create these histos only once (as a data members of the digitizer) and then fill/reset them.

Reply:
Done